src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1018,6 +1021,21 @@ public static Double count (Object o) {
     }
 
     /**
+     * A node is considered non-empty if it is convertible into a string with a greater-than zero length.
+     *
+     * @param o NodeSet to evaluate. Throws if not a NodeSet
+     * @return the number of non-empty nodes in argument node-set.
+     */
+    public static Double countNonEmpty (Object o) {
```

Comment:
This should return `int`.  It’s not the concern of this method that its caller must return a reference variable. A discrete count should be an integer.

Reply:
Yes, both great points. This one should be private.

Reply:
Sounds good, again though I wanted to be consistent with the method above: `public static Double count(Object)`. Should I change that method to be `private int` or create an issue as a follow up PR to make these changes so the two methods that provide counting are consistent?